js.go

```diff
@@ -2766,11 +2756,214 @@ func (sub *Subscription) Fetch(batch int, opts ...PullOpt) ([]*Msg, error) {
 	}
 	// If there is at least a message added to msgs, then need to return OK and no error
 	if err != nil && len(msgs) == 0 {
-		return nil, checkCtxErr(err)
+		return nil, o.checkCtxErr(err)
 	}
 	return msgs, nil
 }
 
+type FetchResult interface {
+	Messages() <-chan *Msg
+	Error() error
```

Comment:
I think we need a way to stamp that the batch is done, (definition of done being that you cannot consume further messages from it), maybe something like:

```suggestion
	Done() bool
```

or maybe context like?

```go
  Done() <-chan struct{}
```

Reply:
Hmm, I would rather go with `Done() bool` - since we are iterating over values, I would like to avoid having to have for loops with `select`, I think boolean is enough - if it's false, you can safely consume messages.